theano/tensor/nnet/abstract_conv.py

```diff
@@ -2247,25 +2242,24 @@ def perform(self, node, inp, out_):
                                 for k_row in xrange(kern.shape[4]):
                                     for k_col in xrange(kern.shape[5]):
                                         # Deciding which kernel to select
-                                        reg_row = (img.shape[2] - 1) - row + \
-                                                  (kern.shape[4] - 1) - k_row
-                                        reg_col = (img.shape[3] - 1) - col + \
-                                                  (kern.shape[5] - 1) - k_col
-                                        reg_row = min(reg_row, kern.shape[2] - 1)
-                                        reg_row = max(reg_row, 0)
-                                        reg_col = min(reg_col, kern.shape[3] - 1)
-                                        reg_col = max(reg_col, 0)
+                                        reg_row = row - ((kern.shape[4] - 1) - k_row)
+                                        reg_col = col - ((kern.shape[45] - 1) - k_col)
+                                        reg_row = max(0, min(reg_row, kern.shape[2] - 1))
+                                        reg_col = max(0, min(reg_col, kern.shape[3] - 1))
```

Comment:
I'm surprised to see max/min here, are you sure this is correct? If a value falls outside the available range, wouldn't you want to ignore it, rather than add it to the nearest row/col?

Isn't this the same as the loops and indices for the [forward convolution](https://github.com/vikramnitin9/Theano/blob/d803e2784a457f71a6eb31c88271c4ca15631d32/theano/tensor/nnet/abstract_conv.py#L1568-L1579), except that the source and target variables are swapped?